dnf/conf/read.py

```diff
@@ -42,8 +43,16 @@ def __iter__(self):
             yield r
 
         # read .repo files from directories specified by conf.reposdir
-        for repofn in (repofn for reposdir in self.conf.reposdir
-                       for repofn in sorted(glob.glob('{}/*.repo'.format(reposdir)))):
```

Comment:
Thanks for rewriting this! I really hate generators with multiple nested for cycles in them.